src/main/java/de/tum/cit/aet/artemis/communication/web/ConversationMessageResource.java

```diff
@@ -203,4 +203,19 @@ public ResponseEntity<Post> updateDisplayPriority(@PathVariable Long courseId, @
         Post postWithUpdatedDisplayPriority = conversationMessagingService.changeDisplayPriority(courseId, postId, displayPriority);
         return ResponseEntity.ok().body(postWithUpdatedDisplayPriority);
     }
+
+    /**
+     * GET /courses/{courseId}/messages/source-posts : Retrieve posts by their IDs
+     *
+     * @param courseId id of the course the posts belong to
+     * @param postIds  list of IDs of the posts to retrieve
+     * @return ResponseEntity with status 200 (OK) containing the list of posts in the response body,
+     *         or with status 400 (Bad Request) if the checks on user, course or post validity fail
+     */
+    @GetMapping("courses/{courseId}/messages/source-posts")
+    @EnforceAtLeastStudent
+    public ResponseEntity<List<Post>> getSourcePostsByIds(@PathVariable Long courseId, @RequestParam List<Long> postIds) {
+        List<Post> posts = conversationMessagingService.getMessageByIds(postIds);
```

Comment:
you are missing a security check that the user is at least student in the course, either change the annoation above or add an additional check here. This should be part of the resource and not be done in a service (except it's clear from the method names)